### ISSUE TITLE
Picom experimental backends crashes on PyCharm intellisense popup

### ISSUE THREAD

**Issue by raember** [yshui/picom#541]
### Platform

Arch Linux 5.8.16.a-1-hardened amd64

### GPU, drivers, and screen setup

GeForce GTX 1660 Ti Mobile
nvidia-dkms 455.38-1
Three monitors side-by-side
```
name of display: :1
display: :1  screen: 0
direct rendering: Yes
Memory info (GL_NVX_gpu_memory_info):
    Dedicated video memory: 6144 MB
    Total available memory: 6144 MB
    Currently available dedicated video memory: 5425 MB
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: GeForce GTX 1660 Ti/PCIe/SSE2
OpenGL core profile version string: 4.6.0 NVIDIA 455.38
OpenGL core profile shading language version string: 4.60 NVIDIA
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6.0 NVIDIA 455.38
OpenGL shading language version string: 4.60 NVIDIA
OpenGL context flags: (none)
OpenGL profile mask: (none)

OpenGL ES profile version string: OpenGL ES 3.2 NVIDIA 455.38
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

### Environment

bspwm

### picom version

vgit-d00c1
I'm using tryone's fork (AUR package: `picom-tryone-git 1483_Next.202.gd00c1c7_2020.10.07-1`) for the dual-kawase filter

### Configuration:
```
#################################
#             Shadows           #
#################################


# Enabled client-side shadows on windows. Note desktop windows
# (windows with '_NET_WM_WINDOW_TYPE_DESKTOP') never get shadow,
# unless explicitly requested using the wintypes option.
#
shadow = true;

# The blur radius for shadows, in pixels. (defaults to 12)
shadow-radius = 12;

# The opacity of shadows. (0.0 - 1.0, defaults to 0.75)
shadow-opacity = 0.4;

# The left offset for shadows, in pixels. (defaults to -15)
shadow-offset-x = -5;

# The top offset for shadows, in pixels. (defaults to -15)
shadow-offset-y = -4;

# Red color value of shadow (0.0 - 1.0, defaults to 0).
# shadow-red = 0

# Green color value of shadow (0.0 - 1.0, defaults to 0).
# shadow-green = 0

# Blue color value of shadow (0.0 - 1.0, defaults to 0).
# shadow-blue = 0

# Hex string color value of shadow (#000000 - #FFFFFF, defaults to #000000). This option will override options set shadow-(red/green/blue)
# shadow-color = "#000000"

# Specify a list of conditions of windows that should have no shadow.
#
# examples:
#   shadow-exclude = "n:e:Notification";
#
# shadow-exclude = []
shadow-exclude = [
	"name = 'Notification'",
	"class_g = 'Conky'",
	"class_g ?= 'Notify-osd'",
	"class_g = 'Cairo-clock'",
	"_GTK_FRAME_EXTENTS@:c",
	"class_g = 'Bspwm'",  # pre-selection overlay
	"class_g = 'Firefox' && argb",  # autoscroll on firefox window
	"name *?= 'rect-overlay'",  # MS Teams screenshare overlay
];

# Specify a X geometry that describes the region in which shadow should not
# be painted in, such as a dock window region. Use
#    shadow-exclude-reg = "x10+0+0"
# for example, if the 10 pixels on the bottom of the screen should not have shadows painted on.
#
# shadow-exclude-reg = ""

# Crop shadow of a window fully on a particular Xinerama screen to the screen.
# xinerama-shadow-crop = false


#################################
#           Fading              #
#################################


# Fade windows in/out when opening/closing and when opacity changes,
#  unless no-fading-openclose is used.
fading = true;

# Opacity change between steps while fading in. (0.01 - 1.0, defaults to 0.028)
fade-in-step = 0.05;

# Opacity change between steps while fading out. (0.01 - 1.0, defaults to 0.03)
fade-out-step = 0.05;

# The time between steps in fade step, in milliseconds. (> 0, defaults to 10)
# fade-delta = 10

# Specify a list of conditions of windows that should not be faded.
# fade-exclude = []

# Do not fade on window open/close.
# no-fading-openclose = false

# Do not fade destroyed ARGB windows with WM frame. Workaround of bugs in Openbox, Fluxbox, etc.
# no-fading-destroyed-argb = false


#################################
#   Transparency / Opacity      #
#################################


# Opacity of inactive windows. (0.1 - 1.0, defaults to 1.0)
# inactive-opacity = 1

# Opacity of window titlebars and borders. (0.1 - 1.0, disabled by default)
frame-opacity = 0.4;

# Let inactive opacity set by -i override the '_NET_WM_OPACITY' values of windows.
inactive-opacity-override = false;

# Default opacity for active windows. (0.0 - 1.0, defaults to 1.0)
# active-opacity = 1.0

# Dim inactive windows. (0.0 - 1.0, defaults to 0.0)
# inactive-dim = 0.0

# Specify a list of conditions of windows that should always be considered focused.
# focus-exclude = []

# Use fixed inactive dim value, instead of adjusting according to window opacity.
# inactive-dim-fixed = 1.0

# Specify a list of opacity rules, in the format `PERCENT:PATTERN`,
# like `50:name *= "Firefox"`. picom-trans is recommended over this.
# Note we don't make any guarantee about possible conflicts with other
# programs that set '_NET_WM_WINDOW_OPACITY' on frame or client windows.
opacity-rule = [
	"90:class_g = 'URxvt'",
	"90:class_g = 'dmenu'",
	"10:class_g = 'Bspwm'", # pre-selection
	"84:class_g = 'Polybar'",
	"60:class_g = 'Dunst'",
	"94:class_g = 'Code'",
	"94:class_g = 'jetbrains-idea'",
	"96:class_g = 'Texmaker'",
	"96:class_g = 'Firefox'",
	"96:class_g = 'discord'",
	"90:class_g = 'whatsapp-desktop'",
	"90:class_g = 'Foxit Reader'",
	"96:class_g = 'GravitDesigner'",
	"94:class_g = 'Spacefm'",
	"90:class_g = 'MATLAB *'",
	"90:class_g = 'Kaku'",
	"96:class_g = 'Spotify'",
	"70:class_g *?= 'Rofi'",
	"30:class_g = 'Escrotum'",
];


#################################
#     Background-Blurring       #
#################################

blur: {
	# Controls the blur method. Corresponds to the --blur-method command line option.
	# Available choices are:
	# 	none to disable blurring;
	# 	gaussian for gaussian blur;
	# 	box for box blur;
	# 	kernel for convolution blur with a custom kernel;
	# 	dual_kawase for dual-filter kawase blur.
	# Note: gaussian, box and dual_kawase blur methods are only supported by the experimental backends.
	# (default: none)
	method = "dual_kawase";

	# The size of the blur kernel, required by gaussian and box blur methods.
	# For the kernel method, the size is included in the kernel.
	# Corresponds to the --blur-size command line option
	# (default: 3).
	#size = 3;

	# The standard deviation for the gaussian blur method.
	# Corresponds to the --blur-deviation command line option
	# (default: 0.84089642).
	#deviation = 0.84089642;

	# The strength of the dual_kawase blur method.
	# Corresponds to the --blur-strength command line option.
	# If set to zero, the value requested by --blur-size is approximated
	# Range: 0-20 (default: 5).
	strength = 8;

	# The kernel to use for the kernel blur method, specified in the same format as the --blur-kerns option.
	# Corresponds to the --blur-kerns command line option.
	# Format:
	# WIDTH,HEIGHT,ELE1,ELE2,ELE3,ELE4,ELE5...
	# OR:
	# 	3x3box (default),
	# 	5x5box,
	# 	7x7box,
	# 	3x3gaussian,
	# 	5x5gaussian,
	# 	7x7gaussian,
	# 	9x9gaussian,
	# 	11x11gaussian
	#kernel = "3x3box";
};

# Blur background of semi-transparent / ARGB windows.
# Bad in performance, with driver-dependent behavior.
# The name of the switch may change without prior notifications.
# (default: false)
blur-background = true;

# Blur background of windows when the window frame is not opaque.
# Implies:
#    blur-background
# Bad in performance, with driver-dependent behavior. The name may change.
# (default: false)
#blur-background-frame = false

# Use fixed blur strength rather than adjusting according to window opacity.
# (default: false)
#blur-background-fixed = false

# Exclude conditions for background blur.
blur-background-exclude = [
#	"window_type = 'dock'",
	"window_type = 'desktop'",
	"_GTK_FRAME_EXTENTS@:c",
#	"class_g = 'Bspwm'", # pre-selection
];

#################################
#       General Settings        #
#################################

# Daemonize process. Fork to background after initialization. Causes issues with certain (badly-written) drivers.
#daemon = false

# Specify the backend to use: `xrender`, `glx`, or `xr_glx_hybrid`.
# `xrender` is the default one.
backend = "glx"

# Enable/disable VSync.
vsync = true;

# Enable remote control via D-Bus. See the *D-BUS API* section below for more details.
#dbus = false

# Try to detect WM windows (a non-override-redirect window with no
# child that has 'WM_STATE') and mark them as active.
mark-wmwin-focused = true;

# Mark override-redirect windows that doesn't have a child window with 'WM_STATE' focused.
mark-ovredir-focused = true;

# Try to detect windows with rounded corners and don't consider them
# shaped windows. The accuracy is not very high, unfortunately.
detect-rounded-corners = true;

# Detect '_NET_WM_OPACITY' on client windows, useful for window managers
# not passing '_NET_WM_OPACITY' of client windows to frame windows.
detect-client-opacity = true;

# Specify refresh rate of the screen. If not specified or 0, picom will
# try detecting this with X RandR extension.
# (default: 60)
refresh-rate = 0;

# Use EWMH '_NET_ACTIVE_WINDOW' to determine currently focused window,
# rather than listening to 'FocusIn'/'FocusOut' event. Might have more accuracy,
# provided that the WM supports it.
#use-ewmh-active-win = false

# Unredirect all windows if a full-screen opaque window is detected,
# to maximize performance for full-screen windows. Known to cause flickering
# when redirecting/unredirecting windows.
#unredir-if-possible = false

# Delay before unredirecting the window, in milliseconds. Defaults to 0.
#unredir-if-possible-delay = 0

# Conditions of windows that shouldn't be considered full-screen for unredirecting screen.
#unredir-if-possible-exclude = []

# Use 'WM_TRANSIENT_FOR' to group windows, and consider windows
# in the same group focused at the same time.
detect-transient = true;

# Use 'WM_CLIENT_LEADER' to group windows, and consider windows in the same
# group focused at the same time. 'WM_TRANSIENT_FOR' has higher priority if
# detect-transient is enabled, too.
detect-client-leader = true;

# Resize damaged region by a specific number of pixels.
# A positive value enlarges it while a negative one shrinks it.
# If the value is positive, those additional pixels will not be actually painted
# to screen, only used in blur calculation, and such. (Due to technical limitations,
# with use-damage, those pixels will still be incorrectly painted to screen.)
# Primarily used to fix the line corruption issues of blur,
# in which case you should use the blur radius value here
# (e.g. with a 3x3 kernel, you should use `--resize-damage 1`,
# with a 5x5 one you use `--resize-damage 2`, and so on).
# May or may not work with *--glx-no-stencil*. Shrinking doesn't function correctly.
#resize-damage = 1

# Specify a list of conditions of windows that should be painted with inverted color.
# Resource-hogging, and is not well tested.
#invert-color-include = []

# GLX backend: Avoid using stencil buffer, useful if you don't have a stencil buffer.
# Might cause incorrect opacity when rendering transparent content (but never
# practically happened) and may not work with blur-background.
# My tests show a 15% performance boost. Recommended.
#glx-no-stencil = false

# GLX backend: Avoid rebinding pixmap on window damage.
# Probably could improve performance on rapid window content changes,
# but is known to break things on some drivers (LLVMpipe, xf86-video-intel, etc.).
# Recommended if it works.
#glx-no-rebind-pixmap = false

# Disable the use of damage information.
# This cause the whole screen to be redrawn everytime, instead of the part of the screen
# has actually changed. Potentially degrades the performance, but might fix some artifacts.
# The opposing option is use-damage
use-damage = true;

# Use X Sync fence to sync clients' draw calls, to make sure all draw
# calls are finished before picom starts drawing. Needed on nvidia-drivers
# with GLX backend for some users.
xrender-sync-fence = true

# GLX backend: Use specified GLSL fragment shader for rendering window contents.
# See `compton-default-fshader-win.glsl` and `compton-fake-transparency-fshader-win.glsl`
# in the source tree for examples.
#glx-fshader-win = ""

# Force all windows to be painted with blending. Useful if you
# have a glx-fshader-win that could turn opaque pixels transparent.
#force-win-blend = false

# Do not use EWMH to detect fullscreen windows.
# Reverts to checking if a window is fullscreen based only on its size and coordinates.
#no-ewmh-fullscreen = false

# Dimming bright windows so their brightness doesn't exceed this set value.
# Brightness of a window is estimated by averaging all pixels in the window,
# so this could comes with a performance hit.
# Setting this to 1.0 disables this behaviour. Requires --use-damage to be disabled. (default: 1.0)
#max-brightness = 1.0

# Make transparent windows clip other windows like non-transparent windows do,
# instead of blending on top of them.
#transparent-clipping = false

# Set the log level. Possible values are:
#  "trace", "debug", "info", "warn", "error"
# in increasing level of importance. Case doesn't matter.
# If using the "TRACE" log level, it's better to log into a file
# using *--log-file*, since it can generate a huge stream of logs.
log-level = "warn";

# Set the log file.
# If *--log-file* is never specified, logs will be written to stderr.
# Otherwise, logs will to written to the given file, though some of the early
# logs might still be written to the stderr.
# When setting this option from the config file, it is recommended to use an absolute path.
#log-file = '/path/to/your/log/file'

# Show all X errors (for debugging)
#show-all-xerrors = false

# Write process ID to a file.
#write-pid-path = '/path/to/your/log/file'

# Window type settings
#
# 'WINDOW_TYPE' is one of the 15 window types defined in EWMH standard:
#     "unknown", "desktop", "dock", "toolbar", "menu", "utility",
#     "splash", "dialog", "normal", "dropdown_menu", "popup_menu",
#     "tooltip", "notification", "combo", and "dnd".
#
# Following per window-type options are available: ::
#
#   fade, shadow:::
#     Controls window-type-specific shadow and fade settings.
#
#   opacity:::
#     Controls default opacity of the window type.
#
#   focus:::
#     Controls whether the window of this type is to be always considered focused.
#     (By default, all window types except "normal" and "dialog" has this on.)
#
#   full-shadow:::
#     Controls whether shadow is drawn under the parts of the window that you
#     normally won't be able to see. Useful when the window has parts of it
#     transparent, and you want shadows in those areas.
#
#   redir-ignore:::
#     Controls whether this type of windows should cause screen to become
#     redirected again after been unredirected. If you have unredir-if-possible
#     set, and doesn't want certain window to cause unnecessary screen redirection,
#     you can set this to `true`.
#
wintypes:
{
	tooltip = {
		fade = true;
		shadow = false;
		opacity = 0.75;
		focus = true;
		full-shadow = false;
	};
	utility = {  # FF autoscroll
		shadow = false;
		blur-background = false;
	};
	dock = { shadow = false; }
	dnd = { shadow = false; }
	popup_menu = { opacity = 0.8; }
	dropdown_menu = { opacity = 0.8; }
};
```

### Steps of reproduction

1. Start picom with `picom --show-all-xerrors  --experimental-backends`
2. Open up PyCharm
3. Start writing a method (`def `)
4. Picom crashes

*probably any JetBrains IDE works as well*

### Expected behavior
No crash

### Current Behavior
Crashes with the following output:
```
[ 13.11.2020 09:42:49.374 win_clear_flags WARN ] Flags cleared on a destroyed window 0x0560420c (win476)
picom: ../src/backend/backend.c:268: paint_all_new: Zusicherung »!(w->flags & WIN_FLAGS_SHADOW_NONE)« nicht erfüllt.
[1]    869987 abort (core dumped)  picom --show-all-xerrors --experimental-backends
```

If experimental backends aren't used (`picom --show-all-xerrors`), it throws the following errors:

```
[ 11/13/20 09:43:26.118 get_cfg WARN ] Dual-kawase blur is not implemented by the legacy backends, you must use the `experimental-backends` option.
[ 11/13/20 09:43:26.203 init_render WARN ] Old backends only support blur method "kernel". Your blur setting will not be applied
[ 11/13/20 09:43:29.488 win_clear_flags WARN ] Flags cleared on a destroyed window 0x05604275 (win477)
[ 11/13/20 09:43:29.492 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.492 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.492 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.494 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.494 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.494 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.503 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.503 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.503 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.522 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.522 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.522 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.538 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.538 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.538 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.555 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.555 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.555 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.572 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.572 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.572 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.590 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.590 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.590 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.607 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.607 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.607 paint_one ERROR ] Window 0x05604275 is missing painting data.
[ 11/13/20 09:43:29.623 glx_bind_pixmap ERROR ] Failed to query info of pixmap 0x08e002d9.
[ 11/13/20 09:43:29.623 paint_one ERROR ] Failed to bind texture for window 0x05604275.
[ 11/13/20 09:43:29.623 paint_one ERROR ] Window 0x05604275 is missing painting data.
```

### Stack trace

```
Starting program: /usr/bin/picom --show-all-xerrors --experimental-backends --debug-mode
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[ 11/13/20 09:47:56.796 win_clear_flags WARN ] Flags cleared on a destroyed window 0x056042d4 (win478)
[ 11/13/20 09:47:56.796 win_clear_flags WARN ] Flags cleared on a destroyed window 0x056042d4 (win478)
picom: ../src/backend/backend.c:268: paint_all_new: Assertion `!(w->flags & WIN_FLAGS_SHADOW_NONE)' failed.

Program received signal SIGABRT, Aborted.
0x00007ffff78ba615 in raise () from /usr/lib/libc.so.6
```

**Comment by tryone144** [comment]
The issue with shadows (assertion failure with `WIN_FLAGS_SHADOW_NONE`) has recently been fixed in #479.
You might want to test again with the current `next` (e.g. using the `picom-git` package from the AUR).

BTW: The dual-filter kawase blur algorithm has been merged into `next`, so there is no need to run my fork anymore. :wink:

**Comment by raember** [comment]
Thank you for your fast response. I should have first looked through the issues instead of opening a new issue. My bad.

Now I also know why I couldn't get dual kawase blur working before: I had used the `picom`package and not the `picom-git`one, which is why I still used your fork. I was puzzled as to why it didn't work even though you stated clearly that dual kawase has been merged into upstream a good while ago. Thanks for clearing that up as well.
